### ISSUE TITLE
Support for lr_scheduler.ReduceLROnPlateau has not been included in version 2.0.1

### ISSUE THREAD

**Issue by bennyjg** [dreamquark-ai/tabnet#231]
**Describe the bug**

version 2.0.1 does not correctly support this scheduler (as far as I can tell).

**What is the current behavior?**

Scheduler ignores loss function and continuously reduces the LR every "patience" epochs.

**If the current behavior is a bug, please provide the steps to reproduce.**

**Expected behavior**

Scheduler should reduce LR only when validation loss starts to grow from minimum for "patience" epochs.
To support this the scheduler should be passed the validation loss ( and not just scheduler.step() )
as I changed in my accepted pull request from previous version.

**Screenshots**

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

The feature (support for ReduceLROnPlateau)
that I added in previous version has not been included in the latest main build.
Do I need to redo the pull request?

**Comment by Optimox** [comment]
Hello @bennyjg,

What makes you think that the scheduler does not work as expected?

This line https://github.com/dreamquark-ai/tabnet/blob/2a9757d4002e7ba6117e53d140581147e651ab24/pytorch_tabnet/callbacks.py#L279 should take care of it

If you plot the learning rates (available with `clf.history`) don't you see the LR reducing when it's supposed to?

**Comment by bennyjg** [comment]
Hi @Optimox,
First, thanks for the clear reply :)
I rechecked my code and found out that I changed the eval_metric to 'auc' which means that I want to maximize the metric and I have to add the option of ' mode = "max" ' to the options for the scheduler to work correctly. So it was my bad that it did not work.
Thanks!
